src/game.cpp

```diff
@@ -5863,40 +5874,54 @@ int game::mon_info(WINDOW *w)
 
 void game::cleanup_dead()
 {
+    // For some reason, using creature_tracker::find here is slow
+    // So we'll be accessing the vector directly, at least for the first pass
+    auto &mon_list = critter_tracker->get_monsters_list();
     // Important: `Creature::die` must not be called after creature objects (NPCs, monsters) have
     // been removed, the dying creature could still have a pointer (the killer) to another creature.
-    for( size_t i = 0; i < num_zombies(); i++ ) {
```

Comment:
Instead of exposing creature_tracker internals, how about pushing this logic down into creature tracker and just expose a creature_tracker::cleanup_dead() method?